scene/2d/navigation_polygon.cpp

```diff
@@ -312,7 +315,7 @@ void NavigationPolygon::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_set_outlines", "outlines"), &NavigationPolygon::_set_outlines);
 	ClassDB::bind_method(D_METHOD("_get_outlines"), &NavigationPolygon::_get_outlines);
 
-	ADD_PROPERTY(PropertyInfo(Variant::POOL_VECTOR3_ARRAY, "vertices", PROPERTY_HINT_NONE, "", PROPERTY_USAGE_NOEDITOR | PROPERTY_USAGE_INTERNAL), "set_vertices", "get_vertices");
+	ADD_PROPERTY(PropertyInfo(Variant::POOL_VECTOR2_ARRAY, "vertices"), "set_vertices", "get_vertices");
```

Comment:
The change to `POOL_VECTOR2_ARRAY` is a bugfix and could be merged right away, but the other changes need review from @reduz or anyone familiar with the notification system.

Reply:
If this is approved, the same should likely be done in 3D for `NavigationMesh`.

Reply:
I added the `POOL_VECTOR3_ARRAY` bugfix in #29285, so the rest should be rebased on top of the current master branch.